### PR TITLE
add check if server returns a 500 and display error

### LINKED ISSUE #17170: Shared folder no access when ldap server is down
Steps to reproduce
1. I have an LDAP server, with some user
2. Entered with the user1 in Oc server (with Ldap)
3. The user 1 create a new folder, call "New folder for share"
4. Go to the folder "New folder for share", and download some files
5. Share this folder with other user (user2) with Ldap
6.throw down the Ldap server

### Expected behaviour
You can go into the Folder

### Actual behaviour
I have two errors:
1. If you go to the user1, and click in the "New folder for share", the name of the folder doubles and does nothing

screen shot 2015-06-25 at 11 40 07

2. And appears the error 500 
screen shot 2015-06-25 at 11 44 53

### Server configuration

DESKTOP v ownCloud-1.8.3.2465-nightly20150625.pkg
Server v {"installed":true,"maintenance":false,"version":"8.1.0.7","versionstring":"8.1 RC1","edition":"Enterprise"}

### PR DESCRIPTION
**Pull request by th3fallen**
resolves https://github.com/owncloud/core/issues/17170

please suggest new string for notification if this isn't adequate @DeepDiver1975 @PVince81

### CODE REVIEW

**Comment by PVince81** [comment]
Sounds good, but I don't think we should redirect to the root in this case. The user might want to retry.

Or is this the case that might cause the infinite redirect loop ?

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -1131,6 +1131,14 @@
 				return false;
 			}
 
+			// Did share service die or something else fail?
+			if (result.status === 500) {
```
Comment (LukasReschke) [inline apps/files/js/filelist.js]: I'd prefer a switch statement.

Comment (th3fallen) [inline apps/files/js/filelist.js]: i initially did it as a switch statement but it got messy quickly.

**Comment by th3fallen** [comment]
> Sounds good, but I don't think we should redirect to the root in this case. The user might want to retry.
> 
> Or is this the case that might cause the infinite redirect loop ?

where would you recommend redirecting to? or what action should be taken?

File: apps/files/js/filelist.js (hunk 1 above)
Comment (LukasReschke) [inline apps/files/js/filelist.js]: :insert_meme_excuse_accepted:

**Comment by PVince81** [comment]
No redirect, just show the notification.

On the other hand it means the file list will appear empty, so might still be fine to redirect.

I'm ok with this change :+1:

File: apps/files/js/filelist.js [hunk 2]
```diff
@@ -1131,6 +1131,14 @@
...
+				// Go home
+				this.changeDirectory('/');
+				OC.Notification.show(t('files', 'This directory is unavailable please check your logs or contact your administrator'));
```
Comment (Xenopathic) [inline apps/files/js/filelist.js]: Needs a comma: '...is unavailable, please check...'.

Also, s/your/the/g (we refer to 'the logs' and 'the administrator' in other places :smile:)

Comment (th3fallen) [inline apps/files/js/filelist.js]: changed

**Comment by LukasReschke** [comment]
Let's get this in :+1:

**Pull request merged**